src/Workspaces/FolderWorkspace_ProjectLoader.cs

```diff
@@ -36,19 +37,15 @@ public virtual async Task<ProjectInfo> LoadProjectInfoAsync(string folderPath, I
                     documents: documents);
             }
 
-            private static Task<ImmutableArray<DocumentInfo>> LoadDocumentInfosAsync(ProjectId projectId, string folderPath, string fileExtension, ImmutableHashSet<string> filesToInclude)
+            private static Task<ImmutableArray<DocumentInfo>> LoadDocumentInfosAsync(ProjectId projectId, string folderPath, string fileExtension, ImmutableHashSet<string> pathsToInclude)
             {
                 return Task.Run(() =>
                 {
-                    string[] filePaths;
-                    if (filesToInclude.Count == 0)
+                    var filePaths = Directory.GetFiles(folderPath, $"*{fileExtension}", SearchOption.AllDirectories);
+                    if (!pathsToInclude.IsEmpty)
                     {
-                        filePaths = Directory.GetFiles(folderPath, $"*{fileExtension}", SearchOption.AllDirectories);
-                    }
-                    else
-                    {
-                        filePaths = filesToInclude.Where(
-                            filePath => filePath.EndsWith(fileExtension) && File.Exists(filePath)).ToArray();
+                        filePaths = filePaths.Where(filePath =>
+                            pathsToInclude.Any(path => filePath.StartsWith(path, StringComparison.OrdinalIgnoreCase))).ToArray();
```

Comment:
Is there a .NET api to compare file paths that works if the file system is case-sensitive (like most linux file systems are?). Not a blocker jsut curious

Reply:
Not that I can find. Nothing off Path at least. Could roll our own getting FileInfo and using its FullPath which I would hope to be cased properly.